funsor/gaussian.py

```diff
@@ -579,11 +579,10 @@ def eager_neg(op, arg):
 
 
 @eager.register(Integrate, Gaussian, Variable, frozenset)
-@integrator
 def eager_integrate(log_measure, integrand, reduced_vars):
     real_vars = frozenset(k for k in reduced_vars if log_measure.inputs[k].dtype == 'real')
-    if real_vars:
-        assert real_vars == frozenset([integrand.name])
+    if real_vars == frozenset([integrand.name]):
```

Comment:
I'd rather error than be lazy here:
```py
if real_vars:
    if real_vars != frozenset([integrand.name]):
        raise NotImplementedError('TODO')
    ...
```
Also it appears there is an error in that int vars is ignored. I believe we should
```py
result = Tensor(data, inputs)
return result.reduce(ops.add, reduced_vars - real_vars)
```

Reply:
> I'd rather error than be lazy here:

The laziness here is necessary to allow `eager` to generically eliminate variables that only appear on one side (i.e. the job currently done by `@integrator`)

Reply:
Hmm, can you explain this in person? I didn't realize we could stage multiple `eager` rules in sequence, which we did with `@integrator`

Reply:
This comes from interleaving `eager` and `normalize` - if an `eager` pattern returns `None` and the resulting term isn't normalized, `normalize` will rewrite it and the expression inside the `normalize` pattern will again be evaluated under `eager`